engine/runtime/src/main/java/org/enso/interpreter/runtime/data/atom/Atom.java

```diff
@@ -304,7 +309,9 @@ final Object readMember(String member, @CachedLibrary(limit = "3") StructsLibrar
     }
     var method = findMethod(member);
     if (method != null) {
-      return method;
+      var ctx = EnsoContext.get(applySelf);
+      var state = ctx.currentState();
+      return applySelf.execute(method, state, new Object[] {this});
```

Comment:
I have hard times understanding what this actually does. Would be nice to see comment: "Returns method with self argument preapplied"

Reply:
C'mon! The node is called _applySelf_. What else do you want? Maybe 14632b2a7360d4b21b7484d86007c734d3d31da8 ?